lib/semantics/resolve-names.cc

```diff
@@ -387,8 +387,10 @@ class ScopeHandler : public ImplicitRulesVisitor {
   std::optional<SubprogramKind> subpNamesOnly_;
 
   // Apply the implicit type rules to this symbol.
-  void ApplyImplicitRules(const SourceName &, Symbol &);
+  void ApplyImplicitRules(Symbol &);
   std::optional<const DeclTypeSpec> GetImplicitType(Symbol &);
+  bool ConvertToObjectEntity(Symbol &);
```

Comment:
Convert what to ObjectEntity?

Reply:
The symbol.

There is a comment on the function definition. I have been putting any comments on the definition rather than the declaration for private functions. In my view the declaration is boilerplate and ignored by human readers. Same with static non-member functions.

Reply:
Not clear to me whether llvm wants comments on the declaration or the definition -- Eric puts most comments on the declaration.  Google is clear that comments about the function interface belong on the declaration and comments about the inner workings of the function belong on the definition.   https://google.github.io/styleguide/cppguide.html#Function_Comments

Reply:
This is a *.cc file.  There is no public API needing documentation in this case.